tools/cli/src/cargo_manifest.rs

```diff
@@ -34,27 +31,67 @@ pub(crate) fn extract_sdk_version(path: &Path) -> Result<Version, ManifestError>
             CargoTomlError::Parse(e) => e.into(),
             CargoTomlError::Io(e) => e.into(),
             CargoTomlError::InheritedUnknownValue => {
-                CannotProcessManifest("inherited unknown value".to_string())
+                ManifestError::CannotProcessManifest("inherited unknown value".to_string())
             }
-            CargoTomlError::WorkspaceIntegrity(reason) => CannotProcessManifest(reason),
-            CargoTomlError::Other(reason) => CannotProcessManifest(reason.to_string()),
-            _ => CannotProcessManifest("Unknown".to_string()),
+            CargoTomlError::WorkspaceIntegrity(reason) => {
+                ManifestError::CannotProcessManifest(reason)
+            }
+            CargoTomlError::Other(reason) => {
+                ManifestError::CannotProcessManifest(reason.to_string())
+            }
+            _ => ManifestError::CannotProcessManifest("Unknown".to_string()),
         }
     })?;
 
-    let dependency = manifest
+    if !manifest.dependencies.contains_key(SKD_CRATE_NAME) {
+        return Err(ManifestError::NonMarineWasm);
+    }
+
+    let package = manifest
+        .package
+        .ok_or_else(|| ManifestError::NoPackageInManifest(format!("{}", path.display())))?;
+
+    // TODO: find & load lockfile once per `marine build` run
+    path.ancestors()
+        .find_map(|path| extract_sdk_version_from_lockfile(&package, &path.join(LOCKFILE_NAME)))
+        .ok_or_else(|| ManifestError::CannotFindLockfile(format!("{}", path.display())))
+}
+
+fn extract_sdk_version_from_lockfile(
+    target_package: &cargo_toml::Package,
+    path: &Path,
+) -> Option<semver::Version> {
+    log::debug!("Trying to load lockfile from {}", path.display());
+    let lockfile = cargo_lock::Lockfile::load(path).ok()?;
+    log::debug!(
+        "Lockfile loaded. Looking for entry for {}@{}",
+        target_package.name.as_str(),
+        target_package.version()
+    );
+    lockfile
+        .packages
+        .iter()
+        .find_map(|package| {
+            if package.name.as_str() == target_package.name.as_str()
+                && &package.version.to_string() == target_package.version()
```

Comment:
Is this comparison enough?

Reply:
Its not. But the only way to do it is to import a crate with same version and name from a different place. I can compare `source` (it is `None` for worspace members). But i dont see benefit from such check

Reply:
Given that a .wasm artifact is required to go to this function, it is highly unlikely situation